### PR TITLE
Module 2 Components

### PR DESCRIPTION
**Pull request by TashaUA**

### CODE REVIEW

File: src/app/modules/components/courses/add-course/add-course.component.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-add-course',
+  templateUrl: './add-course.component.html',
+  styleUrls: ['./add-course.component.scss']
+})
+export class AddCourseComponent implements OnInit {
+
+  constructor() { }
```
Comment (goremchik) [inline src/app/modules/components/courses/add-course/add-course.component.ts]: Remove empty constructors and life cycle hooks

File: src/app/modules/components/courses/course-item/course-item.component.ts [hunk 2]
```diff
@@ -1,15 +1,31 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
+import { Course } from "../../../../core/models/course";
```
Comment (goremchik) [inline src/app/modules/components/courses/course-item/course-item.component.ts]: Create alias to core folder and use smth like this: `@core/models/course`

File: src/app/modules/components/courses/course-list/course-list.component.html [hunk 3]
```diff
@@ -1 +1,8 @@
-<p>course-list works!</p>
+<div class="courses">
+  <app-course-item
+      *ngFor="let course of courses; index as i; trackBy: courseTrackBy"
+      [course]="course"
+      (courseChangeEvent)="onCourseChange($event)"
+  ></app-course-item>
+  <div class="courses__load-more"><a href="#" (click)="onLoadMore($event)">Load More</a></div>
```
Comment (goremchik) [inline src/app/modules/components/courses/course-list/course-list.component.html]: let's add it as button, but not link

File: src/app/modules/components/courses/course-list/course-list.component.ts [hunk 4]
```diff
@@ -7,9 +8,53 @@ import { Component, OnInit } from '@angular/core';
 })
 export class CourseListComponent implements OnInit {
 
+  public courses: Course[] = [
+    {
```
Comment (goremchik) [inline src/app/modules/components/courses/course-list/course-list.component.ts]: move this to mock folder

File: src/app/modules/components/courses/search/search.component.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    border-color: #ced0db;
+    border-width: 1px;
+    border-style: solid;
+    height: 36px;
+    width: 240px;
+    padding: 0 0 0 10px;
+  }
+
+  &__button {
+    border: 1px solid #67A300;
```
Comment (goremchik) [inline src/app/modules/components/courses/search/search.component.scss]: Create constants file and import it in all files

File: src/app/modules/components/courses/add-course/add-course.component.ts (hunk 1 above)
Comment (TashaUA) [reply src/app/modules/components/courses/add-course/add-course.component.ts]: done

File: src/app/modules/components/courses/course-item/course-item.component.ts (hunk 2 above)
Comment (TashaUA) [reply src/app/modules/components/courses/course-item/course-item.component.ts]: done

File: src/app/modules/components/courses/course-list/course-list.component.html (hunk 3 above)
Comment (TashaUA) [reply src/app/modules/components/courses/course-list/course-list.component.html]: fixed

File: src/app/modules/components/courses/course-list/course-list.component.ts (hunk 4 above)
Comment (TashaUA) [reply src/app/modules/components/courses/course-list/course-list.component.ts]: moved

File: src/app/modules/components/courses/search/search.component.scss (hunk 5 above)
Comment (TashaUA) [reply src/app/modules/components/courses/search/search.component.scss]: added

**Pull request merged**